koku/reporting_common/models.py

```diff
@@ -111,3 +113,20 @@ class Meta:
 
     region = models.CharField(max_length=32, null=False, unique=True)
     region_name = models.CharField(max_length=64, null=False, unique=True)
+
+
+class SourceServiceProduct(models.Model):
```

Comment:
I see that this table provides some flexibility to not re-write the materialized views if we ever change out the products we are considering for network or database, but I think it also causes some issues for compute and storage since we don't look at product type for those today. 

See: https://github.com/project-koku/koku/blob/e32b5d9c8724d19db5d1d79a09eb4210a1d3e67f/koku/api/report/all/openshift/provider_map.py#L265-L268 

and

https://github.com/project-koku/koku/blob/e32b5d9c8724d19db5d1d79a09eb4210a1d3e67f/koku/api/report/all/openshift/provider_map.py#L421